### PR TITLE
PI-230 Create incident report page and form (UI only)

### PR DESCRIPTION
**Pull request by utainwitedsonthi**
image

### CODE REVIEW

File: src/html/modules/report/about.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<div class="container">
```
Comment (rassokhin-s) [inline src/html/modules/report/about.html:1]: 1) `src/html/modules/report/about.html` is not correct name for the file's purpose. It contains the form, so I would name it `form.html`
2) If you take a look at other pages and files, you will see that we use more precise word for naming the root class in the file (not `container`)

File: src/html/modules/report/about.html [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        </select>
+      </div>
+    </div>
+    <!-- Comments -->
+    <div class="container_row">
+      <div class="container_column_25">
+        <label for="comments">Comments</label>
+      </div>
+      <div class="container_column_75">
+        <textarea id="comments" name="comments" style="height:200px"></textarea>
```
Comment (rassokhin-s) [inline src/html/modules/report/about.html:104]: Please avoid using inline styles in this repo. Each html file has its own styles file where you can put all your styles code.

File: src/less/modules/report/report.less [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  &_column_20 {
+    float: left;
+    width: 20%;
+    margin-top: 10px;
+  }
+
+  &_column_25 {
+    float: left;
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:32]: Once you read about BEM methodology for styles, you will find out that you can do the following:
```
.form {
  &__column {
    float: left;
    margin-top: 10px;
    &_20 {
      width: 20%;
    }
    &_40 {
      width: 40%;
    }
  }
}
```

File: src/less/modules/report/report.less [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  &_row::after {
+    content: "";
+    display: table;
+    clear: both;
+  }
+}
+
+label {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:56]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  padding: 12px 12px 12px 0;
+  display: inline-block;
+}
+
+input[type=text],
+select,
+textarea {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:63]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  width: 100%;
+  padding: 12px;
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  resize: vertical;
+}
+
+input[type=date] {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:71]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  width: 100%;
+  padding: 12px 12px 12px 0;
+  display: inline-block;
+}
+
+input[type=time] {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:77]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  width: 40%;
+  padding: 12px 12px 12px 0;
+  display: inline-block;
+}
+
+input[type=url] {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:83]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+  width: 100%;
+  padding: 12px 12px 12px 0;
+  display: inline-block;
+}
+
+input[type=submit] {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:89]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+  padding: 20px 40px;
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+  display: block;
+  margin: 0 auto;
+  margin-top: 20px;
+}
+
+input[type=submit]:hover {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:102]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+  background-color: #45a049;
+}
+
+.to-time {
+  display: inline-block;
+  margin: 5px;
+}
+
+textarea {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:111]: Don't apply styles to tag unless you are sure that it won't affect other components on the page.
Please make it with classname instead.

File: src/less/modules/report/report.less [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+  resize: none;
+}
+
+@media screen and (max-width: 1200px) {
+
+  .container_column_10,
+  .container_column_20,
+  .container_column_25,
+  .container_column_40,
+  .container_column_75 {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:121]: You can write it shorter with BEM methodology and Less.

File: src/less/modules/report/report.less [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+input[type=url] {
+  width: 100%;
+  padding: 12px 12px 12px 0;
+  display: inline-block;
+}
+
+input[type=submit] {
+  background-color: #4CAF40;
+  color: white;
+  font-size: large;
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:92]: This one will look different for different users...

File: src/less/modules/report/report.less [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+  padding: 20px 40px;
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+  display: block;
+  margin: 0 auto;
+  margin-top: 20px;
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:99]: These two lines can be replaced with one:
```
margin: 20px auto 0;
```

File: src/less/modules/report/report.less [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    margin-top: 10px;
+  }
+
+  &_column_75 {
+    float: left;
+    width: 75%;
+    margin-top: 10px;
+  }
+
+  &_row::after {
```
Comment (rassokhin-s) [inline src/less/modules/report/report.less:49]: We already have `clearfix` mixing in this repo. Please use it.

**Review by rassokhin-s — CHANGES_REQUESTED** [review]
In addition to comments which I left for the code:
1) The UI design looks not very clear for the first view. Labels column on the left side looks liked a page sidebar. Please make it look more like a form.
2) We have a contact form on the homepage. I think they should have similar design. So either we make this form look similar, or we change the homepage form to have new design.
3) `select` and `input` elements have different border colors.

File: src/less/modules/report/form.less [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+    &_40 {
+      width: 40%;
+      float: left;
+      margin-top: 10px;
+    }
+
+    &_75 {
+      width: 75%;
+      float: left;
+      margin-top: 10px;
```
Comment (rassokhin-s) [inline src/less/modules/report/form.less:30]: You don't need to repeat `float: left; margin-top: 10px;` for every of `&_20`, `&_25`, `&_40`, `&_75`. You can put it one time in the following place:
```
&__column {
  float: left;
  margin-top: 10px;
```

File: src/less/modules/report/report.less (hunk 13 above)
Comment (rassokhin-s) [reply src/less/modules/report/report.less]: This is not fixed yet.

**Review by rassokhin-s — CHANGES_REQUESTED** [review]
There is still two small change requests.

**Comment by rassokhin-s** [comment]
https://user-images.githubusercontent.com/2122991/104292585-10654300-54ce-11eb-98b9-cd91e56bf280.mov
A few other comments:
1) I'm not able to open calendar, date and time pickers by clicking on them - only by clicking on the icon. It should open when I click on any place of the input.
2) When hover over dropdown, calender, date or time picker, the cursor should be "pointer".
3) When I open calendar or date or time picker it uses blue color as highlight. We don't have this color in our branding. Please use green instead.

**Comment by antonyharfield** [comment]
We will not merge this and instead make it part of the new incident response dashboard.

**Pull request closed without merging**